service/lib/agama/storage/manager.rb

```diff
@@ -135,7 +135,7 @@ def probe(keep_config: false, keep_activation: true)
           _("Calculating the storage proposal")
         )
 
-        product_config.pick_product(software.selected_product)
+        product_config.pick_product(software.config["product"])
```

Comment:
The Ruby tests fail in places where the software client was switched from D-Bus to HTTP. Please fix that, the test file already contains similar mocks for other HTTP services:

service/test/agama/storage/manager_test.rb